acts

```diff
@@ -161,6 +146,20 @@ backuprc=0 # Notice any failed backups
 for dir in $backuptargets; do
     archive_starttime=$(date +%s)
     nicedirname=$(echo "$dir" | tr -d '/')
+    # Determine the archive type to create
+    if echo "$archives" | grep -E -q "^$hostname-yearly-$year-.+-$nicedirname"; then
```

Comment:
The pattern (and all others that now have `.+` in the middle should end with `\$`, just to be safe.